Model/SecondChanceRepository.php

```diff
@@ -187,10 +272,249 @@ public function delete(SecondChanceInterface $SecondChance)
+                ['eq' => $store->getId()]
+            )
+            ->addFieldToFilter('created_at', ['lteq' => new \Zend_Db_Expr('NOW() - INTERVAL ' . $timing . ' DAY')])
+            ->addFieldToFilter('created_at', ['gteq' => new \Zend_Db_Expr('NOW() - INTERVAL 5 DAY')]);
+
+        foreach ($collection as $item) {
+            $order = $this->orderFactory->create()->loadByIncrementId($item->getOrderId());
+
+            //BP-896 skip Transfer method
+            $payment = $order->getPayment();
+            if (in_array($payment->getMethod(), [\Buckaroo\Magento2\Model\Method\Transfer::PAYMENT_METHOD_CODE, \Buckaroo\Magento2\Model\Method\PayPerEmail::PAYMENT_METHOD_CODE])) {
+                $this->setFinalStatus($item, $final_status);
+                continue;
+            }
 
-        return $this->delete($SecondChance);
+            if ($item->getLastOrderId() != null &&
+                $last_order = $this->orderFactory->create()->loadByIncrementId($item->getLastOrderId())) {
+                if ($last_order->hasInvoices()) {
+                    $this->setFinalStatus($item, $final_status);
+                    continue;
+                }
+            }
+
+            if ($order->hasInvoices()) {
+                $this->setFinalStatus($item, $final_status);
+            } else {
+                if ($this->accountConfig->getNoSendSecondChance($store)) {
+                    $this->logging->addDebug(__METHOD__ . '|getNoSendSecondChance|');
+                    if ($this->checkOrderProductsIsInStock($order)) {
+                        $this->logging->addDebug(__METHOD__ . '|checkOrderProductsIsInStock|');
+                        $this->sendMail($order, $item, $step);
+                    }
+                } else {
+                    $this->logging->addDebug(__METHOD__ . '|else getNoSendSecondChance|');
+                    $this->sendMail($order, $item, $step);
+                }
+            }
+        }
+        $collection->save();
+    }
+
+    public function sendMail($order, $secondChance, $step)
+    {
+        $this->logging->addDebug(__METHOD__ . '|sendMail start|');
+        $configProvider = $this->configProviderFactory->get('second_chance');
+        $config         = $configProvider->getConfig();
+
+        $store          = $order->getStore();
+        $vars           = [
+            'order'                    => $order,
+            'billing'                  => $order->getBillingAddress(),
+            'payment_html'             => $this->getPaymentHtml($order),
+            'store'                    => $store,
+            'formattedShippingAddress' => $this->getFormattedShippingAddress($order),
+            'formattedBillingAddress'  => $this->getFormattedBillingAddress($order),
+            'secondChanceToken'        => $secondChance->getToken(),
+        ];
+
+        if ($step == 1) {
+            $templateId = $this->scopeConfig->getValue($config['template'], ScopeInterface::SCOPE_STORE, $store) ??
```

Comment:
I've see a config provider added, that should be used here too. and scopeconfig could be removed in this class.

Reply:
fixed